offix/src/main/java/org/aerogear/offix/interceptor/OffixConflictCallback.kt

```diff
@@ -0,0 +1,52 @@
+package org.aerogear.offix.interceptor
+
+import android.util.Log
+import com.apollographql.apollo.api.Error
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+import org.aerogear.offix.*
+import org.json.JSONObject
+
+/*
+OffixConflictCallback
+ */
+class OffixConflictCallback : ApolloInterceptor.CallBack {
+    private val TAG = javaClass.simpleName
+
+    override fun onResponse(response: ApolloInterceptor.InterceptorResponse) {
+
+        /* Check if the conflict is present in the response of not using the ConflictResolutionHandler class.
+         */
+        if (ConflictResolutionHandler().conflictPresent(response.parsedResponse)) {
+
+            /* Parse the response into a json object from the server and extract the serverState and clientState.
+               Make an object of ServerClientdata and add to the list.
+            */
+            val extensions = JSONObject(
+                ((response.parsedResponse.get().errors()[0] as Error).customAttributes()["extensions"] as Map<*, *>)["exception"] as Map<*, *>
+            )
+
+            val exception = extensions.optJSONObject("conflictInfo")
+            val serverState = exception.optJSONObject("serverState")
+            val clientState = exception.optJSONObject("clientState")
+
+            Log.d("$TAG 2", extensions.toString())
+            Log.d("$TAG 3", serverState.toString())
+            Log.d("$TAG 4", clientState.toString())
+
+            list.add(ServerClientData(serverState.toString(), clientState.toString()))
```

Comment:
Simply return the serverClientData; don't maintain a list since you only get the 0th element anyways.

Reply:
*HashMap<>